### ISSUE TITLE
[java] AbstractClassWithoutAnyMethod - false negative with class having a constructor only

### ISSUE THREAD

**Issue by vanguard-1024** [pmd/pmd#4045]
**Affects PMD Version:6.47.0** 

**Rule:AbstractClassWithoutAnyMethod**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_design.html#abstractclasswithoutanymethod>

**Description:**
This rule is used to detect abstract class which is like a data container, I found a false negative in the following code case. This code case is a data container, but PMD doesn't report the warning. I think a warning in line 1 is necessary.

**Code Sample demonstrating the issue:**

```java
public abstract class CC {  // should repoart a warning here
    String data;
    public CC(String data) {
        this.data = data;
    }
}
```

**Expected outcome:**

PMD should report a violation at line 1, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by jsotuyod** [comment]
The class has a constructor and is therefore intentionally whitelisted by the rule. The rule states:

> If an abstract class does not provide any methods, it may be acting as a simple data container that is not meant to be instantiated. In this case, it is probably better to use a private or protected constructor in order to prevent instantiation than make the class misleadingly abstract.

In this case, as a constructor is provided, instantiation is clearly intended. The constructor may even not only initialize data, but have side effects, so all the more reason to accept it.

Why do you consider this case should be flagged?

**Comment by adangel** [comment]
Closing this as invalid. If you have more information, feel free to comment.
